src/diffusers/models/transformers/transformer_z_image_omni.py

```diff
@@ -0,0 +1,1005 @@
+            ]
+        )
+        self.t_embedder = TimestepEmbedder(min(dim, ADALN_EMBED_DIM), mid_size=1024)
+        self.cap_embedder = nn.Sequential(RMSNorm(cap_feat_dim, eps=norm_eps), nn.Linear(cap_feat_dim, dim, bias=True))
+        self.siglip_embedder = nn.Sequential(RMSNorm(siglip_feat_dim, eps=norm_eps), nn.Linear(siglip_feat_dim, dim, bias=True))
+
+        self.x_pad_token = nn.Parameter(torch.empty((1, dim)))
+        self.cap_pad_token = nn.Parameter(torch.empty((1, dim)))
+        self.siglip_pad_token = nn.Parameter(torch.empty((1, dim)))
+
+        self.layers = nn.ModuleList(
+            [
+                ZImageTransformerBlock(layer_id, dim, n_heads, n_kv_heads, norm_eps, qk_norm)
+                for layer_id in range(n_layers)
+            ]
+        )
+        head_dim = dim // n_heads
+        assert head_dim == sum(axes_dims)
+        self.axes_dims = axes_dims
+        self.axes_lens = axes_lens
+
+        self.rope_embedder = RopeEmbedder(theta=rope_theta, axes_dims=axes_dims, axes_lens=axes_lens)
+
+    def unpatchify(
+        self,
+        unified: List[torch.Tensor],
+        size: List[Tuple],
+        patch_size,
+        f_patch_size,
+        x_pos_offsets,
+    ) -> List[torch.Tensor]:
+
+        pH = pW = patch_size
+        pF = f_patch_size
+        bsz = len(unified)
+        assert len(size) == bsz
+
+        x = []
+        for i in range(bsz):
+            x_item = []
+            unified_x = unified[i][x_pos_offsets[i][0]:x_pos_offsets[i][1]]
+            cu_len = 0
+            for j in range(len(size[i])):
+                if size[i][j] is None:
+                    x_item.append(None)
+                    ori_len = 0
+                    pad_len = SEQ_MULTI_OF
+                    cu_len += (pad_len + ori_len)
+                else:    
+                    F, H, W = size[i][j]
+                    ori_len = (F // pF) * (H // pH) * (W // pW)  # without padding
+                    pad_len = (-ori_len) % SEQ_MULTI_OF
+                    # assert ori_len + pad_len == unified_x.shape[0], f"Batch item {i}, patch {j}: ori_len {ori_len} + pad_len {pad_len} != unified_x.shape[0] {unified_x.shape[0]}"
+                    x_item.append(einops.rearrange(
+                        unified_x[cu_len:cu_len + ori_len].view(F // pF, H // pH, W // pW, pF, pH, pW, self.out_channels),
+                        "f h w pf ph pw c -> c (f pf) (h ph) (w pw)",
+                    ))
+                    cu_len += (ori_len + pad_len)
+            x.append(x_item[-1])
+        return x
+
+    @staticmethod
+    def create_coordinate_grid(size, start=None, device=None):
+        if start is None:
+            start = (0 for _ in size)
+
+        axes = [torch.arange(x0, x0 + span, dtype=torch.int32, device=device) for x0, span in zip(start, size)]
+        grids = torch.meshgrid(axes, indexing="ij")
+        return torch.stack(grids, dim=-1)
+
+    def patchify_and_embed(
```

Comment:
this method is really hard to follow here, do you think it's possible to break it into 3?

like


```py
for x, cap_feat, siglip_feat in zip(all_x, all_cap_feats, all_siglip_feats):
    cap_item_cu_len = 1
    
    cap_padded, ..., cap_item_cu_len = self.patchify_and_embed_cap(...)
    all_cap_padded.append(cap_padded)
    
    x_padded, ..., cap_item_cu_len = self.patchify_and_embed_x(..., cap_item_cu_len)
    all_x_padded.append(x_padded)
    ...
    
    siglip_padded, ..., cap_item_cu_len = self.patchify_and_embed_siglip(...,cap_item_cu_len )
    all_siglip_padded.append(siglip_padded)
 ```